// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

syntax = "proto2";

package quickstep.serialization;

import "types/Type.proto";
import "types/TypedValue.proto";

message Comparison {
  enum ComparisonID {
    EQUAL = 0;
    NOT_EQUAL = 1;
    LESS = 2;
    LESS_OR_EQUAL = 3;
    GREATER = 4;
    GREATER_OR_EQUAL = 5;
    LIKE = 6;
    NOT_LIKE = 7;
    REGEX_MATCH = 8;
    NOT_REGEX_MATCH = 9;
  }

  required ComparisonID comparison_id = 1;
}

message OperationSignatureLite {
  required string op_name = 1;
  repeated TypeID param_type_ids = 2;
  repeated bool param_is_static = 3;
}

message OperationSignature {
  required OperationSignatureLite sig_lite = 1;
  repeated Type param_types = 2;
  repeated TypedValue static_param_values = 3;
}
